File: mobile/mycrewmanager/lib/features/notification/domain/entities/notification.dart
import 'package:equatable/equatable.dart';

class Notification extends Equatable {
  final int id;
  final String type;
  final String title;
  final String message;
  final bool isRead;
  final DateTime createdAt;
  final String? actionUrl;
  final String? actor;
  final int? objectId;

  const Notification({
    required this.id,
    required this.type,
    required this.title,
    required this.message,
    required this.isRead,
    required this.createdAt,
    this.actionUrl,
    this.actor,
    this.objectId,
  });

  @override
  List<Object?> get props => [
        id,
        type,
        title,
        message,
        isRead,
        createdAt,
        actionUrl,
        actor,
        objectId,
      ];
}

File: backend-node/src/models/Notification.js
import mongoose from 'mongoose';

const NOTIFICATION_TYPES = [
  'project_invitation', 'task_assigned', 'task_updated', 'task_completed',
  'mention', 'deadline_reminder', 'project_update', 'project_status_changed',
  'member_joined', 'member_left', 'task_due_date_set',
];

const notificationSchema = new mongoose.Schema(
  {
    recipient: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    notificationType: { type: String, enum: NOTIFICATION_TYPES, required: true },
    title: { type: String, required: true },
    message: { type: String, required: true },
    contentObject: { type: mongoose.Schema.Types.Mixed, default: null },
    isRead: { type: Boolean, default: false },
    readAt: { type: Date, default: null },
    actionUrl: { type: String, default: null },
    actor: { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
  },
  { timestamps: true, collection: 'notifications' }
);

notificationSchema.index({ recipient: 1, isRead: 1 });
notificationSchema.index({ recipient: 1, createdAt: -1 });
notificationSchema.index({ createdAt: 1 }, { expireAfterSeconds: 90 * 24 * 60 * 60 }); // TTL 90 days

export const Notification = mongoose.model('Notification', notificationSchema);
